languagetool-language-modules/de/src/main/java/org/languagetool/rules/de/GermanSpellerRule.java

```diff
@@ -136,12 +136,14 @@ public class GermanSpellerRule extends CompoundAwareHunspellRule {
-  private static final Pattern INVALID_COMP_PART = compile("kontrolle|perspektive|schule|sprache|stelle|suche");
+  private static final Pattern NEEDS_TO_BE_PLURAL = compile("adresse|aktie|antenne|apache|arbeitnehmer(in)?|ärztin|astronom(in)?|autor(in)?|azteke|bakterie|ballade|bauer|billion|bisexuelle|blume|börse|buche|bürg(e|in)|bürokrat(in)?|chrysantheme|dän(e|in)?|debatte|decke|diakon(in)?|domäne|drohne|druid(e|in)?|düne|ehre|eibe|ellipse|emittent(in)?|elfe|elle|enge|erbse|eremit|erde|erste|esche|fabrikant(in)?|falke|fassade|farbe|felge|ferien|figur|fluor|frage|frau|förde|galle|gerät|gezeit|gilde|göttin|halt|heid(e|in)?|herde|historie|hölle|höhle|hose|hund|jesuit|jungfer|kante|kaskade|kathode|katze|kette|kid|klasse|kirche|klaue|klinge|knappe|koeffizient|kojote|komödie|kontrahent|konfirmand(in)?|krake|kralle|kranke|krähe|kraut|krippe|kuriosität|kurve|kusine|küste|laie|laterne|laute|legende|lehne|leise|lerche|leuchte|lippe|loge|lotse|länge|läuse|löwe|lücke|luke|made|maske|maßnahme|matriarchin|menge|mensch|metapher|methode|metropole|miene|miete|million|mineral|miniatur|mitte|maus|monarch(in)?|mormone|mücke|mühle|musikant(in)?|mysterium|nerv|niederlage|nixe|nonne|note|obdachlose|ode|organist|panne|parzelle|pate|patient|patriarch(in)?|petze|pfanne|pfeife|platte|polle|pomade|pomeranze|posse|praktikant(in)?|prise|prominente|prototyp|prälat|puppe|pädophile|rabe|radikale|rakete|rampe|ranke|rate|raupe|rendite|repressalie|rest|riese|rinde|rind|robbe|robe|romanist|rose|ross|route|nummer|runde|röhre|rübe|salbe|schabe|schale|scheide|schelle|schenke|schere|sphäre|dicke|kröte|schlampe|schlange|schluchte|schmiere|schnake|schnalle|schneide|schnelle|schokolade|schotte|schurke|schwabe|schwalbe|schwede|schwule|seele|seide|seite|serie|silbe|sonne|sorge|sorte|spanne|sparte|sperre|spitze|sproße|spule|steppe|straße|streife|studie|stunde|stütze|tabelle|tinte|tote|toilette|torte|traube|treffe|treppe|truhe|träne|tunte|tüte|urne|vene|versicherte|verwandte|virtuose|vorname|waffe|wanne|ware|watte|wehe|welle|wiese|zentrum");
```

Comment:
There are some words where the plural form would not (always) be right for compounds, e.g., _miniatur_ (_Miniaturausgabe, Mi­ni­a­tur­bild, Mi­ni­a­tur­ge­mäl­de, Miniaturgolf_ etc.) and also words that would rather be shortened than used in plural in compounds, like _leuchte -> leucht (Leuchtmittel), metropole -> metropol (Metropolregion), mineral -> mineral (Mineralwasser)_, _radikal -> radikal (Radikaldiät)_ etc.

Reply:
yes, similar to `adresse` -> Adressbuch etc.

Reply:
>There are some words where the plural form would not (always) be right for compounds

There are a lot of such words in German. It is extremely time-consuming to collect an exhaustive list for those cases. That's why I decided to go for either 'always plural' or 'always singular' for less common cases. If you think 'This word is more often singular than plural when used in compounds' or the other way around, I can change the 'default numerus'. 

But creating lists that differantiate between singular and plural for words (e. g. Figurproblem vs Figurengedicht), is really time-consuming. I did this for common words that sometimes require infix s and sometime don't allow it, but this becomes a Sisyphean task at some point.

> also words that would rather be shortened than used in plural in compounds
Words like 'Leuchtmittel' are already accepted by other parts of the code. But we should also allow words like 'Leuchtenhersteller', but we should still flag typos like 'Leuchtemittel' or 'Leuchtehersteller'. This is what the code aims at.